values_test.go

```diff
@@ -23,15 +23,15 @@ func Test_date(t *testing.T) {
 	require.Regexp(t, "^[0-9]{4}(-[0-9]{2}){2}T([0-9]{2}:){2}[0-9]{2}Z$", v)
 }
 
-func TestGetValues(t *testing.T) {
+func TestGetFlags(t *testing.T) {
 	// prepare the repo
 	repo := newRepo(t)
 	defer os.RemoveAll(repo.dir)
 	mkCommit(t, repo, "commit 1")
 	mkCommit(t, repo, "commit 2")
 
 	// read the flags
-	fl, err := GetFlags(repo.dir, "main")
+	fl, err := GetFlags(repo.dir, []string{})
```

Comment:
can you make it work with `nil` (and pass `nil` too). nil should be a substitute for empty slice.

Reply:
Actually, I believe that the test should be passing something akin to `[]string{"./govvv", "build", ...}`.  

I can see the value in facilitating `nil` being a substitute for an empty slice in a robust API.  However, I'm not sure that it's applicable here and, actually, was going to recommend changing the visibility of `GetFlags()` to be private.

I don't think others will be using a `GetFlags()` function that's stuck in the `main` package.  If the function did provide compelling value, then imho it should go into the `govvv` package and return a struct:

    type GovvvVersions struct {
        Version    string
        BuildDate  time.Time
        GitCommit  string
        GitBranch  string
        GitState   string
        GitSummary string
    }

jm2c

Reply:
As I think more on it, `[]string{"./govvv", "build", ...}` obfuscates the unit tests and the `GetFlags()` function gets a thorough workout via the integration tests.  Using the same reasoning, I believe that using `nil` instead would obfuscate the unit tests as well.

Reply:
I mean we shouldn't export GetFlags that's for sure. But I mean that in Go, functions that work with empty slice should work with nil slice as well. It's not an invalid value from a language idioms perspective.